stack-graphs/src/stitching.rs

```diff
@@ -719,9 +720,9 @@ impl PathStitcher {
 /// [`find_all_complete_partial_paths`]: #method.find_all_complete_partial_paths
 pub struct ForwardPartialPathStitcher {
     candidate_partial_paths: Vec<Handle<PartialPath>>,
-    queue: VecDeque<PartialPath>,
-    next_iteration: VecDeque<PartialPath>,
-    cycle_detector: CycleDetector<PartialPath>,
+    queue: VecDeque<(PartialPath, JoiningCycleDetector)>,
+    next_iteration: (VecDeque<PartialPath>, VecDeque<JoiningCycleDetector>),
```

Comment:
The frequency with with we're passing these tuples around tells me that the cycle detector might fit better as a field of `PartialPath`.  There's a wrinkle that there are separate `Appending` and `Joining` cycle detectors, but maybe they could be combined?